### DISCUSSION TITLE
add support for glide / portamento

### DISCUSSION THREAD

**Discussion by rghvdberg** [sfztools/sfizz#1211] [Requests]
I've had a request for "glide" (I thought that was the same as portamento :blush: )
> Is there a Glide, Pan L / R and "prelisten samples by click" function in Drops? and should I add it to the git ?
> because these are really the only three Features I miss on it!

Seems others would welcome that option too

> Yes please! A straightforward retrig/legato mono option. And separately glide/auto glide - that way we can enjoy polyphonic portamento as well. And separately per sample or per group - have you tried a stack of unison oscillators each with slightly different portamento times? It’s lovely.

**Comment by paulfd** [comment]
It's not super clear to me what exact functionality does this relates to. So let's have examples:
- A "glide" could be a monophonic sample instrument with say a loop. I play a note, and a legato note after, and basically instead of "choking" the previous note, it smoothly changes its pitch.
- Polyphonic I don't know how that would work, which note do you steal there?
- ???

Maybe some examples would be helpful :)

**Comment by MTornPDX** [comment]
Glide (or portamento) is probably best approached separately from legato modes. In monophonic playback (wether legato or not), it simply slides from the previous pitch to the new one. There are some fine details on how that slide can be calculated. It's useful both when retriggering the sample and EGs (retrig mode) and in legato.
Auto glide (usually an option) will skip the whole glide operation if the new note isn't played legato (a gap between it and the previous note).

Polyphonic glide might be a bit tricky in SFZ, since it doesn't have a set polyphony number. It usually is used in an analog synth connect where you have a specific number of voices (typically 6 or 8) that's being cycled through. So when a note is triggered, the glide would happen from the pitch of the oldest note, wether it's stolen and interrupted by the new note, or completed and not currently being played at all. Different analog polys use different note-cycling (and note stealing) schemes, so the poly glide varies as well. It's always a little chaotic, but often useful and fun.

Most sampler plugins have some of these functions, your best bet is to play around with those. Or better yet, a decent vintage analog synth emulator plugin.

**Comment by paulfd** [comment]
There's support for cc140 now, which allows for some crude portamento effects apparently. I'll track this in another release as this seems more complex, and would probably require extensions to the SFZ opcode set.
